lib/matrix.js

```diff
@@ -52,11 +58,351 @@ var init = function(exports){
+                var userId = this.credentials.userId;
+                var members = this.store.getStateEvents(roomId, 'm.room.member')
+                    .filter(function(event) {
+                        return event.event.user_id !== userId;
+                    });
+                
+                if (members.length == 0) {
+                    return "Unknown";
+                }
+                else if (members.length == 1) {
+                    return members[0].event.content.displayname || members[0].event.user_id;
+                }
+                else if (members.length == 2) {
+                    return (members[0].event.content.displayname || members[0].event.user_id) + " and " +
+                           (members[1].event.content.displayname || members[1].event.user_id);
+                }
+                else {
+                    return (members[0].event.content.displayname || members[0].event.user_id) + " and " +
+                           (members.length - 1) + " others";
+                }
+            }
+        },
+        
+        /*
+         * Helper method for retrieving the name of a user suitable for display in the UI
+         * in the context of a room - i.e. disambiguating from any other users in the room.
+         * XXX: This could perhaps also be generated serverside, perhaps by just passing 
+         * a 'disambiguate' flag down on membership entries which have ambiguous displaynames?
+         */
+        getFriendlyDisplayName: function(userId, roomId) {
+            // we need a store to track the inputs for calculating display names
+            if (!this.store) return userId;
+            
+            var displayName;
+            var memberEvent = this.store.getStateEvent(roomId, 'm.room.member', userId);
+            if (memberEvent && memberEvent.event.content.displayname) {
+                displayName = memberEvent.event.content.displayname;
+            }
+            else {
+                return userId;
+            }
+            
+            var members = this.store.getStateEvents(roomId, 'm.room.member')
+                .filter(function(event) {
+                    return event.event.content.displayname === displayName;
+                });
+                
+            if (members.length > 1) {
+                return displayName + " (" + userId + ")";
+            }
+            else {
+                return displayName;
+            }
+        },
+
+        /*
+         * High level helper method to call initialSync, emit the resulting events,
+         * and then start polling the eventStream for new events.
+         */
+        startClient: function(callback, historyLen) {
+            historyLen = historyLen || 12;
+            
+            var self = this;
+            if (!this.fromToken) {
+                this.initialSync(historyLen, function(err, data) {
+                    if (err) {
+                        console.error("err %s", JSON.stringify(err));
```

Comment:
We shouldn't make too many assumptions on how this SDK should be used. People may have their own logging libraries setup which log in `callback(err)`. I don't propose removing this per se, just that error logging should be wrapped around an optional ``config.debug`` flag or similar.